common/service.go

```diff
@@ -0,0 +1,85 @@
+// Copyright 2017 Microsoft. All rights reserved.
+// MIT License
+
+package common
```

Comment:
This is basically a duplicate of the base plugin class renamed to service. We should really move it to your directory, rest of the code doesn't need it.